src/Servers/Kestrel/Core/src/Internal/Http2/Http2MessageBody.cs

```diff
@@ -46,16 +45,29 @@ protected override void OnDataRead(long bytesRead)
             AddAndCheckConsumedBytes(bytesRead);
         }
 
-        public static MessageBody For(Http2Stream context)
+        public static MessageBody For(Http2Stream context, Http2MessageBody previousMessageBody)
         {
             if (context.ReceivedEmptyRequestBody)
             {
                 return ZeroContentLengthClose;
             }
```

Comment:
Since the above logic was moved to Http2Stream.CreateMessageBody, do we need it here? I wouldn't want to accidentally  pool ZeroContentLengthClose.